core/ecschema-locaters/src/SchemaFileLocater.ts

```diff
@@ -162,7 +251,41 @@ export abstract class SchemaFileLocater {
    * @param matchType The SchemaMatchType to use when comparing the desired Key and the keys found during the search.
    * @param format The type of file that the schema key refers to. json or xml
    */
-  private addCandidateSchemaKeys(foundFiles: FileSchemaKey[], schemaPath: string, fileFilter: string, desiredKey: SchemaKey, matchType: SchemaMatchType, format: string) {
+  private async addCandidateSchemaKeys(foundFiles: FileSchemaKey[], schemaPath: string, fileFilter: string, desiredKey: SchemaKey, matchType: SchemaMatchType, format: string) {
+    const fullPath = path.join(schemaPath, fileFilter);
+
+    const result = new glob.GlobSync(fullPath, { sync: true });
+    for (const match of result.found) {
+      let fileName = path.basename(match, (`.ecschema.${format}`));
+      // TODO: should this be moved or handled elsewhere?
+      // Handles two version file names - SchemaKey.parseString supports only 3 version names.
+      if (/[^\d]\.\d?\d\.\d?\d$/.test(fileName)) {
+        const parts = fileName.split(".");
+        parts.splice(2, 0, "00");
+        fileName = parts.join(".");
+      }
```

Comment:
This code is the same between sync and async methods.  Pull it into a common method to avoid duplication